src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -261,6 +261,7 @@ var addEventListenerForSearchKeyboard = () => {
       // also allow Escape key to hide (but not show) the dynamic search field
       else if (document.activeElement === input && /Escape/i.test(event.key)) {
         toggleSearchField();
+        resetSearchAsYouTypeResults();
```

Comment:
If the user types a query into the search box and then presses `Esc` to close the modal, then the last inline search results should be removed.